packages/default-field-editors/src/Field.tsx

```diff
@@ -0,0 +1,196 @@
+import * as React from 'react';
+import { NumberEditor } from '@contentful/field-editor-number';
+import { SingleLineEditor } from '@contentful/field-editor-single-line';
+import { BooleanEditor } from '@contentful/field-editor-boolean';
+import { DateEditor } from '@contentful/field-editor-date';
+import { LocationEditor } from '@contentful/field-editor-location';
+import { JsonEditor } from '@contentful/field-editor-json';
+import { MultipleLineEditor } from '@contentful/field-editor-multiple-line';
+import { TagsEditor } from '@contentful/field-editor-tags';
+import { SlugEditor } from '@contentful/field-editor-slug';
+import { DropdownEditor } from '@contentful/field-editor-dropdown';
+import { UrlEditor } from '@contentful/field-editor-url';
+import { RadioEditor } from '@contentful/field-editor-radio';
+import { RatingEditor } from '@contentful/field-editor-rating';
+import { CheckboxEditor } from '@contentful/field-editor-checkbox';
+import { ListEditor } from '@contentful/field-editor-list';
+import {
+  SingleEntryReferenceEditor,
+  MultipleEntryReferenceEditor,
+  SingleMediaEditor,
+  MultipleMediaEditor,
+} from '@contentful/field-editor-reference';
+import { RichTextEditor } from '@contentful/field-editor-rich-text';
+import { MarkdownEditor } from '@contentful/field-editor-markdown';
+import type { FieldExtensionSDK } from '@contentful/field-editor-shared';
+import '@contentful/field-editor-date/styles/styles.css';
+import type { WidgetType } from './types';
+
+type FieldProps = {
+  sdk: FieldExtensionSDK;
+  widgetId: WidgetType;
+  isInitiallyDisabled: boolean;
+  renderFieldEditor?: (
+    widgetId: WidgetType,
+    sdk: FieldExtensionSDK,
+    isInitiallyDisabled: boolean
```

Comment:
I am wondering, wouldn't this be a much nicer api? should also save us some code in web creator as we just want to inject a couple of props in some editors. 

```ts
  renderFieldEditor?: (
    widgetId: WidgetType,
    DefaultEditorComponent: React.Component,
    defaultEditorProps: Object // So `return <DefaultEditorComponent {...defaultEditorProps} />` should have the same effect than `return false;`
  ) => JSX.Element | false;
```

On the other hand, with this approach one had to be super careful as `DefaultEditorComponent` might change from version to version and therefore potentially the `defaultEditorProps`

Reply:
I see, good suggestion. There's no way to type the props however, so it would require checking the code/types of editor you want to render

Reply:
Also this would not allow to render something completely custom? Or that's not the case to support here?

Reply:
> you'd need to extract sdk from `defaultEditorProps`

`defaultEditorProps` might not have `sdk`. I was thinking since you instantiated `Field` which takes an `sdk` in the first place you should already have it at your disposal :shrug: But might not hurt to make it available anyways.

> There's no way to type the props however, so it would require checking the code/types of editor you want to render

Yes, that's what I meant by you would need to be "user careful". I think for that reason my idea actually doesn't make much sense, you would be like

```js
<Field {...someProps} renderFieldEditor={
  ({ widgetId, DefaultEditorComponent, defaultEditorProps }) => {
    if (sdk.field.type === FIELD_TYPE_I_WANT_TO_CUSTOMIZE) {
      if (DefaultEditorComponent === COMPONENT_I_WOULD_EXPECT_HERE) {
        return <DefaultEditorComponent {...defaultEditorProps} someProp={SOME_OVERRIDE_FOR_THIS_PROP} />
      } else {
        return <COMPONENT_I_WOULD_EXPECT_HERE {...ALL_THE_PROPS} />
      }
    }
  }
/>

```

**IN SHORT:** Let's forget about this, it was a terrible idea.